### DISCUSSION TITLE
Add configurable punishment to players that failed a quest

### DISCUSSION THREAD

**Discussion by TowardtheStars** [ejektaflex/Bountiful#213] [Mod Ideas]
When playing on server, there are players who just keep a lot of bounties and wait till them get expired. That's kind of annoying for other players who want to get bounties. Also, there is no way for ops to track those players. Besides, a little punishment can chase players to complete bounties, instead of "Meh, I can just throw it away whenever I want."

So maybe add configurable punishment for players who failed to complete bounties? For example, preventing them from obtaining bounties in a configurable period of time. Also, an executable command to punish players can do the work.

**Comment by ejektaflex** [comment]
What kind of result are you looking for?  Unfortunately once a player puts a bounty in a chest, it's out of my control what happens to the bounty. I cannot make it so that something occurs once the bounty expires - because the bounty actually does not "tick", ever. Instead, it just records the time when the bounty was taken and how long the bounty has to complete, and thus when hovering over it / when turning in the quest, it can calculate whether you completed the bounty in time.

Because of this, the only option would be to have said configurable option get triggered when in the player's inventory because I can check for items in a player's inventory, but once they place it in a chest, as I'd said, it's out of my control what happens to it. 

Do you have any ideas @TowardtheStars ?

**Comment by TowardtheStars** [comment]
Maybe adding an additional inventory to players to store all bounty items can do the work. And its GUI should allow players to:
- Never take bounty out of that special inventory 
  - Maybe just don't add players' vanilla inventory slots in GUI I think?
  - Also we need to prevent players from dropping items to the world.
- View bounties
  - Hover mouse on the slot
- Abort bounties 
  - Adding an extra abort slot and players can just drag bounty item into it
  - Or just hold a hotkey while clicking on bounty items
- Check if they have accomplished them and get rewards
  - Left click on the bounty slot

By doing so, we can assure that a player can never transmit bounty items into a chest or give them to other players. And item ticking problem solved.

**Comment by ejektaflex** [comment]
A separate bounty inventory per player could definitely work, and does make sense. I'm working on Bountiful v3 again, and this would be something I'd implement after everything else but I really like the idea 👍

**Comment by ejektaflex** [comment]
@TowardtheStars I think I've found a simple solution - not allowing a single user to take more than a specified amount per X minutes. Which would take a little work, but I could try it out for a v3.1 release after 3.0 is out. How's that sound?

**Comment by TowardtheStars** [comment]
Sounds nice.

**Comment by No3371** [comment]
Configurable deposit seems more viable.

Say a very rare and rewarding bounty requires to pay 1 diamond to take, and that diamond will be refund upon bounty completion.
